DEV-INTRO.md

```diff
@@ -1,59 +1,76 @@
 
-#### List of modules 
+### First time setup
 
-Installing PEcAn
-----------------
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
 
-### Virtual Machine
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
 
-* Fastest way to get started
-* see PEcAn demo ...
+Next we will create the folders that will hold all the data for the docker containers using: `mkdir -p volumes/{lib,pecan,portainer,postgres,rabbitmq,traefik}`. The `volumes` folder will be ignored by git.
 
-### Installing from source
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
 
-#### From GitHub
+During the time it takes for the database to start we will copy the R packages from a container to our `volumes/lib` folder. This folder contain the compiled R packages. Later we will put our newly compiled code here as well. You can copy all the data using `docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -r /usr/local/lib/R/site-library/* /rlib/`. This will copy all compiled packages to your local machine. This only needs to be done once (or if the PEcAn base image changes drastically, for example a new version of R).
 
-```
-library(devtools)
-install_github("pecan", "PEcAnProject")
-```
+At this point postgresql should have started and is ready to load the database. This is done using `docker run -ti --rm --network pecan_pecan pecan/db`. This only needs to be done once (unless the volumes folder is removed).
 
-#### "Makefile"
+Once the database is loaded we can bring up the rest of the docker stack using `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d`. At this point you have PEcAn running in docker.
 
-```
-./scripts/build.sh -install # installs all R packages
-./scripts/build.sh -h       # list other options
+### PEcAn Development
 
-```
\ No newline at end of file
+The current folder is mounted in some containers as `/pecan`. You can see which containers exactly in `docker-compose.dev.yaml`. You can now modify the code on your local machine, or you can use [rstudio](http://localhost:8000) in the docker stack. Once you made changes to the code you can compile the code either in the terminal of rstudio (`cd pecan && make`) or using `./scripts/compile.sh` from your machine (latter is nothing more than a shell script that runs `docker-compose exec executor sh -c 'cd /pecan && make'`. 
```

Comment:
```suggestion
The current folder is mounted in some containers as `/pecan`. You can see which containers exactly in `docker-compose.dev.yaml`. You can now modify the code on your local machine, or you can use [rstudio](http://localhost:8000/rstudio/) in the docker stack. Once you made changes to the code you can compile the code either in the terminal of rstudio (`cd pecan && make`) or using `./scripts/compile.sh` from your machine (latter is nothing more than a shell script that runs `docker-compose exec executor sh -c 'cd /pecan && make'`. 
```

Reply:
I'm also curious if these will cause any permission errors would the code compiled in rstudio (1000:1000) be accessible by web (33:33)

Reply:
volumes are mounted from your laptop and will be owned by the user that is inside the container, in case of rstudio it will be user with id 1001 (rstudio) in case of executor it will be user with id 0 (root).

Not merging suggest since rstudio runs at` http://localhost:8000/rstudio/`

Reply:
hmm I think I still have to worry about permission issues if I follow this correctly. On my laptop this is fine, but on the server I am not root, and when I run pecan I want my user to own those files. But again this doesn't need to hold this PR.

Also, I'm confused, my suggestion was to change the url from `http://localhost:8000` to `http://localhost:8000/rstudio/`? I also added a slash before `cd /pecan` for rstudio terminal command

Reply:
on the server you will need to add your user to the docker group `adduser istfer docker`

Reply:
sounds good to me, thank you.

Reply:
added hint about adding yourself to docker group.